gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -427,6 +476,48 @@ public void delete(URI path, boolean recursive) throws IOException {
     repairImplicitDirectory(parentInfoFuture);
   }
 
+  private String getBucketName(URI path) {
+    if (path.toString() == null || path.toString() == "") return "";
+    int ind = path.toString().replace("gs://", "").indexOf("/");
+    return path.toString().replace("gs://", "").substring(0, ind);
+  }
+
+  private String getFolderName(URI path) {
+    if (path.getPath() == null || path.getPath() == "") return "";
+    return path.getPath().substring(path.getPath().indexOf("/") + 1);
+  }
+
+  /**
+   * Returns the list of folder resources in the prefix. It lists either all the folder resources or
+   * lists at max of 1 folder resource other than prefix
```

Comment:
Not sure if this is accurate about this function. I guess it will depend on the listFolderOptions?

Reply:
This comment is now updated.